src/libAtomVM/externalterm.c

```diff
@@ -268,12 +319,20 @@ static int serialize_term(uint8_t *buf, term t, GlobalContext *glb)
         int atom_index = term_to_atom_index(t);
         size_t atom_len;
         atom_ref_t atom_ref = atom_table_get_atom_ptr_and_len(glb->atom_table, atom_index, &atom_len);
-        if (!IS_NULL_PTR(buf)) {
-            buf[0] = ATOM_EXT;
-            WRITE_16_UNALIGNED(buf + 1, atom_len);
-            atom_table_write_bytes(glb->atom_table, atom_ref, atom_len, buf + 3);
+        AtomString atom_str = atom_table_get_atom_string(glb->atom_table, atom_index);
```

Comment:
That's too bad, because `atom_table_write_bytes` requires that I allocate a buffer, right?  (whereas `atom_table_get_atom_string` just returns the AtomString in the hash table)

Reply:
As soon as we don't have `atom_table_get_atom_string` anymore around, we can enable safe code replace and unload for older versions. But that's another topic.

Reply:
I am not sure what you mean by this.